src/wallet/bdb.cpp

```diff
@@ -656,16 +656,14 @@ void BerkeleyDatabase::ReloadDbEnv()
     env->ReloadDbEnv();
 }
 
-bool BerkeleyBatch::StartCursor()
+BerkeleyCursor::BerkeleyCursor(BerkeleyDatabase& database)
 {
-    assert(!m_cursor);
-    if (!pdb)
-        return false;
-    int ret = pdb->cursor(nullptr, &m_cursor, 0);
-    return ret == 0;
+    assert(database.m_db.get());
+    int ret = database.m_db->cursor(nullptr, &m_cursor, 0);
+    assert(ret == 0);
```

Comment:
Being annoying once again, but should probably replace this with an exception? (see discussion at https://github.com/bitcoin/bitcoin/pull/24914/commits/2c7b214cbc06ac07311add940925683c779e49b4#r893736353 which was marked as resolved, but was actually never done)

Reply:
link is broken, ref https://github.com/bitcoin/bitcoin/pull/24914#discussion_r893736353

Reply:
Done now.